charts/authentik/templates/worker/deployment.yaml

```diff
@@ -38,15 +38,19 @@ spec:
         {{- end }}
         {{- end }}
     spec:
+      {{- if .Values.worker.serviceAccount.create }}
+      {{- if .Values.worker.serviceAccount.name }}
+      serviceAccountName: {{ .Values.worker.serviceAccount.name }}
+      {{- else }}
+      serviceAccountName: {{ template "authentik.server.fullname" . }}
+      {{- end }}      
+      {{- else }}      
       {{- with .Values.worker.imagePullSecrets | default .Values.global.imagePullSecrets }}
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
-      {{- end }}
-      {{- with .Values.worker.serviceAccountName }}
+      {{- end }}            
+      {{- with .Values.server.serviceAccountName }}
```

Comment:
Same as the server. Also this references server.serviceAccountName although we're templating the worker deployment here